bindings/rust/s2n-tls/src/raw/connection.rs

```diff
@@ -307,10 +324,25 @@ impl Connection {
         Ok(self)
     }
 
+    pub(crate) fn set_pending_callback(&mut self, callback: Option<Box<dyn AsyncCallback>>) {
+        self.context_mut().pending_callback = callback;
+    }
```

Comment:
if you need to set multiple callbacks, I think this is going to overwrite the callback and also leak the previous callback

Reply:
`pending_callback` isn't the callback configured for the connection / config to use later: that's still stored on the config. This is the callback that s2n_negotiate has actually triggered and is currently blocked on. We can only block on one callback at a time.

I'll add a debug_assert to enforce that and use `Option::insert` just in case.